examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -582,6 +587,10 @@ RegularGridDecomposition::categorizeParticlesIntoLeftAndRightNeighbor(const std:
         const auto periodicPosition = position[direction] - globalBoxDimensions[direction];
         position[direction] = std::max(_globalBoxMin[direction], periodicPosition);
         rightNeighborParticles.back().setR(position);
+        // Shifting the rebuild position of the particle, so that displacement since rebuild remains consistent
```

Comment:
This is actually quite important and should definitely be mentioned somewhere in the user documentation because it will impact performance if ignored.
Also please check the SPH examples for this.

Reply:
done: added the comment by Sam, user documentation for dynamic rebuild as discussed, and changes in SPH code